packages/console/src/ds-components/Select/index.tsx

```diff
@@ -58,6 +70,50 @@ function Select<T extends string>({
     event.stopPropagation();
   };
 
+  const getSearchInputContainerStyles = () => {
+    if (!anchorRef.current) {
+      return {};
+    }
+    const element = anchorRef.current;
+    const cs = getComputedStyle(element);
+    const paddingX = Number.parseFloat(cs.paddingLeft) + Number.parseFloat(cs.paddingRight);
+    const paddingY = Number.parseFloat(cs.paddingTop) + Number.parseFloat(cs.paddingBottom);
+    const borderX = Number.parseFloat(cs.borderLeftWidth) + Number.parseFloat(cs.borderRightWidth);
+    const borderY = Number.parseFloat(cs.borderTopWidth) + Number.parseFloat(cs.borderBottomWidth);
+    return {
+      width: `${element.offsetWidth - paddingX - borderX}px`,
+      height: `${element.offsetHeight - paddingY - borderY}px`,
+      top: `${
+        element.getBoundingClientRect().top +
+        Number.parseFloat(cs.borderTopWidth) +
+        Number.parseFloat(cs.paddingTop)
+      }px`,
+      left: `${
+        element.getBoundingClientRect().left +
+        Number.parseFloat(cs.borderLeftWidth) +
+        Number.parseFloat(cs.paddingLeft)
+      }px`,
+      backgroundColor: cs.backgroundColor,
+    };
+  };
+
+  const resizeHandler = () => {
+    setWidth(window.innerWidth);
+  };
+
+  useEffect(() => {
+    if (!isOpen) {
+      setSearchInputValue('');
+    }
+  }, [isOpen]);
+
+  useEffect(() => {
+    window.addEventListener('resize', resizeHandler);
+    return () => {
+      window.removeEventListener('resize', resizeHandler);
+    };
+  }, []);
```

Comment:
We have a `useWindowResize` hook to help simplify this resize handling stuff.

Reply:
Just checked the code, and if the `width` state is no where to be used, we might not need this window resize listening stuff any more.